### PR TITLE
Avoid installing pre-release by default

### PR DESCRIPTION
**Pull request by FriggaHel**
Changes:
- Avoid installing pre-release except if mentioned
- Always skip drafts
- Skip versions with no assets

### CODE REVIEW

File: dist/main/index.js [hunk 1]
```diff
@@ -16376,8 +16384,13 @@ async function selectCompatibleVersion(versionSpec) {
     
     const versions = response.data;
     for (let i = 0; i < versions.length; i++) {
-        if (versionSpec === undefined || versionSpec === "latest"
-            || semver.satisfies(versions[i].tag_name, versionSpec)) {
+        if (versions[i].draft || versions[i].assets?.length === 0) {
```
Comment (FriggaHel) [inline dist/main/index.js:16387]: Note: I used a three `if` approach as it gives a much better readability compared to one unique `if` with multiple cases.

File: src/install.js [hunk 2]
```diff
@@ -41,8 +49,13 @@ async function selectCompatibleVersion(versionSpec) {
     
     const versions = response.data;
     for (let i = 0; i < versions.length; i++) {
-        if (versionSpec === undefined || versionSpec === "latest"
-            || semver.satisfies(versions[i].tag_name, versionSpec)) {
+        if (versions[i].draft || versions[i].assets?.length === 0) {
```
Comment (alexplischke) [inline src/install.js:52]:
```suggestion
        if (versions[i].draft) {
```

I think there's always assets, no? I thought GitHub by default includes the source code zip/tar.gz files as part of a release? Maybe that check is unnecessary then.

Comment (FriggaHel) [reply src/install.js:52]: On the website, it looks as an asset.
In the API this is a dedicated field.

=>  https://api.github.com/repos/saucelabs/saucectl-run-action/releases

I'll suggest to keep for the edge case of "Release binaries being built latency".

Comment (alexplischke) [reply src/install.js:52]: Ohh! Good to know 👍

File: src/install.js [hunk 3]
```diff
@@ -22,6 +22,14 @@ function getPlatform() {
     return osName && arch && `${osName}_${arch}`;
 }
 
+function needsDefaultVersion(versionSpec) {
```
Comment (alexplischke) [inline src/install.js]: Maybe more apt?

```suggestion
function isLatest(versionSpec) {
```

Comment (FriggaHel) [reply src/install.js]: You're right, that's more self-explanatory ! I will update that.

**Review by alexplischke — APPROVED** [review]

**Review by tianfeng92 — APPROVED** [review]

**Pull request merged**